chain/client/src/stateless_validation/partial_witness/partial_witness_tracker.rs

```diff
@@ -48,13 +48,21 @@ impl RsMap {
         self.rs_map
             .entry(total_parts)
             .or_insert_with(|| {
-                let data_parts = std::cmp::max((total_parts as f32 * RATIO_DATA_PARTS) as usize, 1);
+                let data_parts = witness_data_parts(total_parts);
                 Arc::new(Some(ReedSolomon::new(data_parts, total_parts - data_parts).unwrap()))
             })
             .clone()
     }
 }
 
+pub fn witness_part_length(encoded_witness_size: usize, total_parts: usize) -> usize {
+    reed_solomon_part_length(encoded_witness_size, witness_data_parts(total_parts))
+}
+
+fn witness_data_parts(total_parts: usize) -> usize {
```

Comment:
nit. function name is a bit ambiguous, maybe change to num_witness_data_parts or num_witness_parts? or add comments.